crates/bevy_ecs/src/query/iter.rs

```diff
@@ -258,7 +258,7 @@ struct QueryIterationCursor<'w, 's, Q: WorldQuery, QF: Fetch<'w, State = Q::Stat
     filter: QueryFetch<'w, F>,
     current_len: usize,
     current_index: usize,
-    phantom: PhantomData<&'w Q>,
+    phantom: PhantomData<(&'w (), Q)>,
```

Comment:
This change shouldn't be needed; what happens if you remove it?

Reply:
@alice-i-cecile Compilation error that I mentioned in Discord. @BoxyUwU suggested this fix for it.

Reply:
for future context `&'w Q` requires `Q: 'w` to hold which is unnecessary, this change makes that not required